<HTML>
<HEAD>
<meta charset="UTF-8">
<title>Validated.Result.Successful.value - validation</title>
<link rel="stylesheet" href="../../../../../style.css">
</HEAD>
<BODY>
<a href="../../../../index.html">validation</a>&nbsp;/&nbsp;<a href="../../../index.html">net.corda.common.validation.internal</a>&nbsp;/&nbsp;<a href="../../index.html">Validated</a>&nbsp;/&nbsp;<a href="../index.html">Result</a>&nbsp;/&nbsp;<a href="index.html">Successful</a>&nbsp;/&nbsp;<a href="./value.html">value</a><br/>
<br/>
<h1>value</h1>
<a name="net.corda.common.validation.internal.Validated.Result.Successful$value(kotlin.Function1((kotlin.collections.Set((net.corda.common.validation.internal.Validated.Result.Successful.ERROR)), java.lang.Exception)))"></a>
<code><span class="keyword">fun </span><span class="identifier">value</span><span class="symbol">(</span><span class="identifier" id="net.corda.common.validation.internal.Validated.Result.Successful$value(kotlin.Function1((kotlin.collections.Set((net.corda.common.validation.internal.Validated.Result.Successful.ERROR)), java.lang.Exception)))/exceptionOnErrors">exceptionOnErrors</span><span class="symbol">:</span>&nbsp;<span class="symbol">(</span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-set/index.html"><span class="identifier">Set</span></a><span class="symbol">&lt;</span><span class="identifier">ERROR</span><span class="symbol">&gt;</span><span class="symbol">)</span>&nbsp;<span class="symbol">-&gt;</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-exception/index.html"><span class="identifier">Exception</span></a><span class="symbol">)</span><span class="symbol">: </span><span class="identifier">TARGET</span></code>
<p>Returns a valid <a href="../../index.html#TARGET">TARGET</a> if no validation errors are present. Otherwise, it throws the exception produced by <a href="../../value.html#net.corda.common.validation.internal.Validated$value(kotlin.Function1((kotlin.collections.Set((net.corda.common.validation.internal.Validated.ERROR)), java.lang.Exception)))/exceptionOnErrors">exceptionOnErrors</a>, defaulting to <a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-illegal-state-exception/index.html">IllegalStateException</a>.</p>
<h3>Exceptions</h3>
<p><a name="IllegalStateException"></a>
<code>IllegalStateException</code> - or the result of <a href="../../value.html#net.corda.common.validation.internal.Validated$value(kotlin.Function1((kotlin.collections.Set((net.corda.common.validation.internal.Validated.ERROR)), java.lang.Exception)))/exceptionOnErrors">exceptionOnErrors</a> if there are errors.</p>
</BODY>
</HTML>
